### DISCUSSION TITLE
How to setup .swcrc file correctly

### DISCUSSION THREAD

**Discussion by jumscrafteur** [swc-project/swc#3475] [Q&A]
Hi, I'm using SWC for the first time and I'm not able to compile my ts files into normal js files.

When I run my command :
```
npx swc ./src -d dist
```
I got the error :
```
failed to process js file

Caused by:
    0: failed to read swcrc file (src/index.ts)
    1: failed to deserialize .swcrc (json) file: syntax error: 1:1
    2: expected value at line 1 column 1
```
Here is my .swcrc I have generated with the [SWC Playground](https://swc.rs/playground)
```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false
    },
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "commonjs"
  },
  "minify": false,
  "isModule": true,
  "sourceMaps": "inline"
}
```
when I run the command with the `--no-swcrc` tag it works but without the .swcrc configurations.

Is there something I don't understand or is my .swcrc file invalid?
Any help would be appreciated 😅

**Comment by ankitchouhan1020** [comment]
Does it work with explicit `--config-file .swcrc` options ?

**Reply by jumscrafteur** [reply]
No, I have the same error as before.

**Reply by ankitchouhan1020** [reply]
Does src/index.ts, typescript file exist ? I'm assuming default config looks for src/index.js

**Reply by jumscrafteur** [reply]
yes, it exists.
here is my entire file hierarchy if it helps:
![image](https://user-images.githubusercontent.com/23343236/152979822-4362bec6-b241-4656-8c24-2367ec229b04.png)
That's what I thought too but the typescript file is well recognized when I use the `--no-swcrc` tag

**Reply by ankitchouhan1020** [reply]
remove `"isModule": true,`

**Reply by jumscrafteur** [reply]
Absolutely no difference 😅

**Reply by ankitchouhan1020** [reply]
https://codesandbox.io/s/vibrant-pare-9fdvy tested on this sandbox. It worked, since isModule is not a valid property.
PS: I cloned this, added your .swcrc file and run your command.

**Reply by jumscrafteur** [reply]
It works! Thanks a lot for your help!
I also needed to change the encoding of my .swcrc file.
By default, it was saved in utf8bom format (and the command didn't work even when removing `isModule`) but in utf8 format, everything works 

I have two last questions:
- Why the the `isModule` property is present in the sandbox? 
![image](https://user-images.githubusercontent.com/23343236/153042133-087ecd74-7175-4952-9334-541b1467da6e.png)
- Is this problem related to the encoding type normal?

**Reply by RiESAEX** [reply]
if you use `utf-8`. it will be like:
    1: failed to deserialize .swcrc (json) file: unmatched data: 17:12
    2: unknown field `isModule`, expected one of `env`, `test`, `exclude`, `jsc`, `module`, `minify`, `inputSourceMap`, `sourceMaps`, `inlineSourcesContent`, `error` at line 17 column 12.
